pkg/secrets/check_rights_nix.go

```diff
@@ -90,3 +91,40 @@ func checkGroupPermission(stat *syscall.Stat_t, usr *user.User, userGroups []str
 
 	return nil
 }
+
+var checkConfigRights = func(path string) error {
+	var stat syscall.Stat_t
+	if err := syscall.Stat(path, &stat); err != nil {
+		return fmt.Errorf("invalid config file '%s': can't stat it: %s", path, err)
```

Comment:
nit: I think `invalid config file` prefix here can be better as the file itself is not "invalid" but its permissions are. Maybe `unable to check permissions on %s` for first one and `invalid config file permissions %s` for the other two?